ush/python/pygfs/task/snowens_analysis.py

```diff
@@ -0,0 +1,368 @@
+
+        # link recentering executable
+        # placeholder, currently already done by the analysis parent class
+
+        # copy fregrid executable
+        fregrid_copy = {'copy': [[os.path.join(self.task_config.EXECgfs, 'fregrid'), os.path.join(self.task_config.DATA, 'fregrid.x')]]}
+        FileHandler(fregrid_copy).sync()
+
+    @logit(logger)
+    def regridDetBkg(self) -> None:
+        """Run fregrid to regrid the deterministic snow background
+        to the ensemble resolution
+
+        Parameters
+        ----------
+        self : Analysis
+           Instance of the SnowEnsAnalysis object
+        """
+
+        chdir(self.task_config.DATA)
+
+        arg_list = [
+            f"--input_mosaic ./orog/det/{self.task_config.CASE}_mosaic.nc",
+            f"--input_dir ./bkg/det/",
+            f"--input_file {to_fv3time(self.task_config.bkg_time)}.sfc_data",
+            f"--scalar_field snodl,slmsk",
+            f"--output_dir ./bkg/det_ensres/",
+            f"--output_file {to_fv3time(self.task_config.bkg_time)}.sfc_data",
+            f"--output_mosaic ./orog/ens/{self.task_config.CASE_ENS}_mosaic.nc",
+            f"--interp_method conserve_order1",
+            f"--weight_file ./orog/det/{self.task_config.CASE}.mx{self.task_config.OCNRES}_oro_data",
+            f"--weight_field land_frac",
+        ]
+        fregrid = os.path.join(self.task_config.DATA, 'fregrid.x') + " " + " ".join(arg_list)
+        exec_cmd = Executable(fregrid)
+
+        try:
+            logger.debug(f"Executing {exec_cmd}")
+            exec_cmd()
+        except OSError:
+            raise OSError(f"Failed to execute {exec_cmd}")
+        except Exception:
+            raise WorkflowException(f"An error occured during execution of {exec_cmd}")
+
+        pass
+
+    @logit(logger)
+    def regridDetInc(self) -> None:
+        """Run fregrid to regrid the deterministic snow increment
+        to the ensemble resolution
+
+        Parameters
+        ----------
+        self : Analysis
+           Instance of the SnowEnsAnalysis object
+        """
+
+        chdir(self.task_config.DATA)
+
+        arg_list = [
+            f"--input_mosaic ./orog/det/{self.task_config.CASE}_mosaic.nc",
+            f"--input_dir ./inc/det/",
+            f"--input_file snowinc.{to_fv3time(self.task_config.bkg_time)}.sfc_data",
+            f"--scalar_field snodl",
+            f"--output_dir ./inc/det_ensres/",
+            f"--output_file snowinc.{to_fv3time(self.task_config.bkg_time)}.sfc_data",
+            f"--output_mosaic ./orog/ens/{self.task_config.CASE_ENS}_mosaic.nc",
+            f"--interp_method conserve_order1",
+            f"--weight_file ./orog/det/{self.task_config.CASE}.mx{self.task_config.OCNRES}_oro_data",
+            f"--weight_field land_frac",
+        ]
+        fregrid = os.path.join(self.task_config.DATA, 'fregrid.x') + " " + " ".join(arg_list)
+        exec_cmd = Executable(fregrid)
```

Comment:
The executable is `fregrid.x` and that should be the `exec_cmd`.  The arguments should be passed to the `exec_cmd` via argument e.g. `exec_cmd(arg_list)` 
I can double check on the `__call__` in `wxflow.Executable`.

Reply:
yeah IIRC I had a lot of trouble getting this to work, so any help/suggestions to try would be appreciated. I know this works, most other things I tried the executable would complain about the arguments being passed to it

Reply:
ok. let me create a test example for you.

Reply:
```python3
fregrid_exe = os.path.join(self.task_config.DATA, 'fregrid.x')
exec_cmd = Executable(fregrid_exe)
exec_cmd(*arg_list)
```
Also added a test for this in wxflow so it serves as a reference for future use.
Please see [here](https://github.com/NOAA-EMC/wxflow/pull/38)

Reply:
Thanks I'll give this a try later today/tomorrow

Reply:
@aerorahul this did not work for me, I'm still getting the issues I was having previously...
```
/scratch1/NCEPDEV/stmp2/Cory.R.Martin/RUNDIRS/snowhighres/esnowrecen.2203935/fregrid.x: unrecognized option '--input_mosaic ./orog/det/C384_mosaic.nc'
/scratch1/NCEPDEV/stmp2/Cory.R.Martin/RUNDIRS/snowhighres/esnowrecen.2203935/fregrid.x: unrecognized option '--input_dir ./bkg/det/'
/scratch1/NCEPDEV/stmp2/Cory.R.Martin/RUNDIRS/snowhighres/esnowrecen.2203935/fregrid.x: unrecognized option '--input_file 20211220.150000.sfc_data'
/scratch1/NCEPDEV/stmp2/Cory.R.Martin/RUNDIRS/snowhighres/esnowrecen.2203935/fregrid.x: unrecognized option '--scalar_field snodl'
/scratch1/NCEPDEV/stmp2/Cory.R.Martin/RUNDIRS/snowhighres/esnowrecen.2203935/fregrid.x: unrecognized option '--output_dir ./bkg/det_ensres/'
/scratch1/NCEPDEV/stmp2/Cory.R.Martin/RUNDIRS/snowhighres/esnowrecen.2203935/fregrid.x: unrecognized option '--output_file 20211220.150000.sfc_data'
/scratch1/NCEPDEV/stmp2/Cory.R.Martin/RUNDIRS/snowhighres/esnowrecen.2203935/fregrid.x: unrecognized option '--output_mosaic ./orog/ens/C192_mosaic.nc'
/scratch1/NCEPDEV/stmp2/Cory.R.Martin/RUNDIRS/snowhighres/esnowrecen.2203935/fregrid.x: unrecognized option '--interp_method conserve_order1'
/scratch1/NCEPDEV/stmp2/Cory.R.Martin/RUNDIRS/snowhighres/esnowrecen.2203935/fregrid.x: unrecognized option '--weight_file ./orog/det/C384.mx025_interp_weight'
/scratch1/NCEPDEV/stmp2/Cory.R.Martin/RUNDIRS/snowhighres/esnowrecen.2203935/fregrid.x: unrecognized option '--weight_field lsm_frac'
/scratch1/NCEPDEV/stmp2/Cory.R.Martin/RUNDIRS/snowhighres/esnowrecen.2203935/fregrid.x: unrecognized option '--remap_file ./remap'
```